src/structs/method.ts

```diff
@@ -366,6 +381,41 @@ ${this.virtualAddress.isNull() ? `` : ` // 0x${this.relativeVirtualAddress.toStr
                 this.fridaSignature
             );
         }
+
+        /** @internal */
+        wrapOnEnter(block: OnEnterCallback): ((this: InvocationContext, args: InvocationArguments) => void) {
+            const startIndex = +!this.isStatic | +Il2Cpp.unityVersionIsBelow201830;
+            return (args: InvocationArguments) => {
+                const thisObject = this.isStatic
+                    ? this.class
+                    : this.class.isValueType
+                    ? new Il2Cpp.ValueType((args[0] as NativePointer).add(Il2Cpp.Object.headerSize - maybeObjectHeaderSize()), this.class.type)
+                    : new Il2Cpp.Object(args[0] as NativePointer);
+                // As opposed to `Interceptor.replace`, `Interceptor.attach` doesn't
+                // interpret pointers, so use `read` instead of `fromFridaValue`
+                const parameters = this.parameters.map((_, i) => read(args[i + startIndex], _.type, false));
+                block.call(thisObject, ...parameters);
+            }
+        }
+
+        /** @internal */
+        wrapOnLeave(block: OnLeaveCallback<T>): ((this: InvocationContext, retval: InvocationReturnValue) => void) {
+            return (retval: InvocationReturnValue) => {
+                // TODO grab `this` pointer during `onEnter`
+                const thisObject = this.class
```

Comment:
Since we derive `this` from the first argument sometimes, I suppose we'll always have to hijack both onEnter and onLeave at the same time, otherwise we can't propagate `this` to `onLeave`. Edit coming up...